File: resources/contacts.go
package resources

import "context"

// Contacts handles contact-related API operations
type Contacts struct {
	client ClientInterface
}

// NewContacts creates a new Contacts resource client
func NewContacts(client ClientInterface) *Contacts {
	return &Contacts{client: client}
}

// ContactSearchParams represents parameters for searching contacts
type ContactSearchParams struct {
	AccountID string `json:"accountID"`
	Query     string `json:"query"`
}

// ContactSearchResponse represents the response from searching contacts
type ContactSearchResponse struct {
	Items []User `json:"items"`
}

// Search searches for contacts/users
func (c *Contacts) Search(ctx context.Context, params ContactSearchParams) (*ContactSearchResponse, error) {
	var result ContactSearchResponse
	queryParams := map[string]interface{}{
		"accountID": params.AccountID,
		"query":     params.Query,
	}
	err := c.client.DoRequestWithQuery(ctx, "GET", "/v0/search-users", queryParams, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

File: beeper-rust-sdk/src/resources/contacts.rs
use crate::client::BeeperDesktop;
use crate::error::Result;
use crate::resources::shared::User;
use reqwest::Method;
use serde::{Deserialize, Serialize};

/// Contacts handles contact-related API operations
#[derive(Debug, Clone)]
pub struct Contacts {
    client: BeeperDesktop,
}

impl Contacts {
    /// Create a new Contacts resource client
    pub fn new(client: BeeperDesktop) -> Self {
        Self { client }
    }

    /// Search searches for contacts/users
    pub async fn search(&self, params: &ContactSearchParams) -> Result<ContactSearchResponse> {
        let query_params = vec![
            ("accountID", params.account_id.as_str()),
            ("query", params.query.as_str()),
        ];

        self.client
            .do_request_with_query(Method::GET, "/v0/search-users", &query_params)
            .await
    }
}

/// ContactSearchParams represents parameters for searching contacts
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ContactSearchParams {
    pub account_id: String,
    pub query: String,
}

/// ContactSearchResponse represents the response from searching contacts
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ContactSearchResponse {
    pub items: Vec<User>,
}